pkgs/development/python-modules/ibis-framework/default.nix

```diff
@@ -60,99 +57,81 @@
-    hash = "sha256-u3BBGdhWajZ5WtoBvNxmx76+orfHY6LX3IWAq/x2/9A=";
+    hash = "sha256-qIxacrRSN/Il9cY6ZsOiYLPyxI20aI+DyDM0/9jr654=";
   };
 
-  patches = [
-    # fixes a small bug in the datafusion backend to reorder predicates
-    (fetchpatch {
-      name = "fix-datafusion-compilation.patch";
-      url = "https://github.com/ibis-project/ibis/commit/009230421b2bc1f86591e8b850d37a489e8e4f06.patch";
-      hash = "sha256-5NHkgc8d2bkOMpbY1vme1XgNfyHSr0f7BrR3JTTjjPI=";
-    })
-  ];
-
   nativeBuildInputs = [
     poetry-core
+    poetry-dynamic-versioning
   ];
 
   propagatedBuildInputs = [
     atpublic
     bidict
+    filelock
     multipledispatch
     numpy
     pandas
     parsy
     pooch
+    pyarrow
     python-dateutil
     pytz
     rich
     sqlglot
     toolz
     typing-extensions
-  ] ++ lib.optionals (pythonOlder "3.9") [ importlib-resources ]
+  ]
   ++ pooch.optional-dependencies.progress
   ++ pooch.optional-dependencies.xxhash;
 
   nativeCheckInputs = [
     pytestCheckHook
-    filelock
     hypothesis
     pytest-benchmark
     pytest-httpserver
     pytest-mock
     pytest-randomly
     pytest-snapshot
     pytest-xdist
-    rsync
   ] ++ lib.concatMap (name: passthru.optional-dependencies.${name}) testBackends;
 
   pytestFlagsArray = [
     "--dist=loadgroup"
     "-m"
     "'${lib.concatStringsSep " or " testBackends} or core'"
-    # sqlalchemy2 breakage
+    # breakage from sqlalchemy2 truediv changes
     "--deselect=ibis/tests/sql/test_sqlalchemy.py::test_tpc_h17"
     # tries to download duckdb extensions
     "--deselect=ibis/backends/duckdb/tests/test_register.py::test_register_sqlite"
     "--deselect=ibis/backends/duckdb/tests/test_register.py::test_read_sqlite"
-    # all the following tests are fixed in the next release (6.0.0)
-    "--deselect=ibis/backends/tests/test_export.py::test_column_to_pyarrow_table_schema"
-    "--deselect=ibis/backends/tests/test_export.py::test_table_to_pyarrow_batches"
-    "--deselect=ibis/backends/tests/test_export.py::test_column_to_pyarrow_batches"
-    "--deselect=ibis/backends/tests/test_export.py::test_to_pyarrow_batches_memtable"
-    "--deselect=ibis/backends/tests/test_export.py::test_table_to_pyarrow_table_schema"
-    "--deselect=ibis/backends/tests/test_export.py::test_to_pyarrow_batches_borked_types"
-    "--deselect=ibis/backends/tests/test_export.py::test_table_pyarrow_batch_chunk_size"
-    "--deselect=ibis/backends/tests/test_export.py::test_column_pyarrow_batch_chunk_size"
-    "--deselect=ibis/backends/tests/test_export.py::test_roundtrip_partitioned_parquet"
-    "--deselect=ibis/tests/sql/test_sqlalchemy.py::test_order_by"
-    "--deselect=ibis/tests/sql/test_sqlalchemy.py::test_no_cart_join"
-    "--deselect=ibis/backends/tests/test_temporal.py::test_temporal_binop"
-    "--deselect=ibis/backends/tests/test_temporal.py::test_interval_literal"
+    # fails with pandas <2
+  ] ++ lib.optionals (lib.versionOlder pandas.version "2.0.0") [
```

Comment:
Do we want to keep this for overrides? We have pandas 2 so we could also drop it.